src/ucm/bistro/bistro.c

```diff
@@ -45,6 +48,59 @@ static ucs_status_t ucm_bistro_protect(void *addr, size_t len, int prot)
     return UCS_OK;
 }
 
+void ucm_bistro_apply_locking(void *dst, const ucm_bistro_lock_t *bytes)
+{
+    uint32_t value;
+
+    if (sizeof(value) > sizeof(*bytes)) {
+        memcpy(&value, dst, sizeof(value));
+    }
+
+    memcpy(&value, bytes, sizeof(*bytes));
+    (void)ucs_atomic_swap32(dst, value);
```

Comment:
does it mean sizeof(*bytes) must be <= sizeof(value) ? can add static assert for it..